### PR TITLE
Added mathematical function jax.numpy.imag

### LINKED ISSUE #18226: imag

### LINKED ISSUE #18981: imag

### PR DESCRIPTION
**Pull request by moezAhmad**
close #18981

### CODE REVIEW

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy/test_math.py [hunk 1]
```diff
@@ -2515,6 +2515,39 @@ def test_jax_conj(
...
+    input_dtype, x = dtype_and_x
+    helpers.test_frontend_function(
+        input_dtypes=input_dtype,
+        test_flags=test_flags,
+        frontend=frontend,
+        fn_tree=fn_tree,
+        on_device=on_device,
+        rtol=1e-2,
+        atol=1e-2,
+        input=x[0],
```
Comment (hmahmood24) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy/test_math.py:2547]: The `input` here should instead be `val`

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy/test_math.py [hunk 2]
```diff
@@ -2515,6 +2515,39 @@ def test_jax_conj(
...
+):
+    input_dtype, x = dtype_and_x
+    helpers.test_frontend_function(
+        input_dtypes=input_dtype,
+        test_flags=test_flags,
+        frontend=frontend,
+        fn_tree=fn_tree,
+        on_device=on_device,
+        rtol=1e-2,
+        atol=1e-2,
```
Comment (hmahmood24) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy/test_math.py:2546]: Are the tests not passing with a low tolerance value? Somewhere around `1e-4`? `1e-2` seems to be too high of a tolerance value

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy/test_math.py [hunk 3]
```diff
@@ -2515,6 +2515,39 @@ def test_jax_conj(
     )
 
 
+# imag
+
+
```
Comment (hmahmood24) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy/test_math.py:2520]: Pretty sure these are lint/formatting issues. Can you kindly resolve these? :) We recommend using `pre-commit` to automatically format your PRs every time you commit. Thanks

**Review by hmahmood24 — CHANGES_REQUESTED** [review]
Hey @moezAhmad, requested some changes for you to look into. Do request a re-review once done. Thanks! :)

**Comment by moezAhmad** [comment]
@hmahmood24 Thank you for reviewing my pr. I have implemented changes as you requested.

**Review by hmahmood24 — APPROVED** [review]
Everything seems good to merge now. Thanks @moezAhmad! 🚀

**Pull request merged**